### PR TITLE
release control from atlas' deadly grip

### PR DESCRIPTION
**Pull request by mauricefallon**
This is the last in the recent series of reorganising PRs (yeah!).
This makes control independent of the atlas types. I have tested this with Drake sim - which works.

The only thing I cannot test is the following:
- ATLAS_STATUS is a giant message published by the driver containing all the ... em ... status
- The fall detector uses the 'behavior' enum to know the high level mode (i.e. is it 'user').
- I have added a message/channel called  drc_behavior/ROBOT_BEHAVIOR for this field
- This message should be published by the driver and used for this.
So, basically, when running Atlas next you'd need to add publishing of this message.
@gizatt  @patmarion : I think its enough to remember this for then

Dependent on:
https://github.com/openhumanoids/atlas-drivers/pull/2
https://github.com/RobotLocomotion/director/pull/169

After this its only the director that is dependent on atlas (types) - and only to display the battery stats and to press "prep" and "stand". If that's resolved the Atlas driver can be removed from the codebase entirely.

### CODE REVIEW

File: software/control/src/CMakeLists.txt [hunk 1]
```diff
@@ -123,7 +123,7 @@ set(java_src_files
   PoseGroundMonitor.java
   RobotPlanPublisher.java
   RobotStateCoder.java
-  AtlasStateExtraCoder.java
+#  AtlasStateExtraCoder.java
```
Comment (patmarion) [inline software/control/src/CMakeLists.txt]: are we not using this file?  should it be deleted?

Comment (mauricefallon) [inline software/control/src/CMakeLists.txt]: I grep'ed the directory and didn't see usage. So  it can be

Funny because, I thought that the arm encoder calibration routine used a matlab instance to capture the end stops - that @mposa wrote.

File: software/control/src/AtlasFallDetector.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ void AtlasFallDetector::findFootIDS() {
 
 void AtlasFallDetector::handleAtlasStatus(const lcm::ReceiveBuffer* rbuf,
                        const std::string& chan,
-                       const atlas::status_t* msg) {
+                       const drc::behavior_t* msg) {
```
Comment (patmarion) [inline software/control/src/AtlasFallDetector.cpp]: I'm confused.  I see some code in this pull request changes atlas::behavior_command_t to drc::behavior_command_t, but this is changing status_t to behavior_t.  What is behavior_t?

Comment (mauricefallon) [inline software/control/src/AtlasFallDetector.cpp]: behavior_t is a message containing only the behavior enum field in status_t
This behavior enum is the high level mode the BDI controller is in.

namely:

struct behavior_t
{
  int64_t utime;
  int32_t behavior;

  const int32_t BEHAVIOR_NONE=0;        //!< Robot currently not in any behavior; software freeze, controls disabled.
  const int32_t BEHAVIOR_FREEZE=1;      //!< All joints frozen in current position.
  const int32_t BEHAVIOR_STAND_PREP=2;  //!< Robot assumes a balanced posture, preparing for Stand.
  const int32_t BEHAVIOR_STAND=3;       //!< Statically stable stand.
  const int32_t BEHAVIOR_WALK=4;        //!< Dynamically stable walk.
  const int32_t BEHAVIOR_STEP=5;        //!< Statically stable, single-step walk.
  const int32_t BEHAVIOR_MANIPULATE=6;  //!< Statically stable stand, with upper body joints controlled by user.
  const int32_t BEHAVIOR_USER=7;        //!< Setpoints and gains set by user.
  const int32_t BEHAVIOR_CALIBRATE=8;   //!< Robot moves through various calibration motions, returns to Freeze.
  const int32_t BEHAVIOR_SOFT_STOP=9;   //!< "Safe" falling behavior, robot tucks its arms and legs and uses soft position gains
}

Comment (mauricefallon) [inline software/control/src/AtlasFallDetector.cpp]: While status_t was a bemoth containing about 25 different types. but this content was irrelevant when drake is controlling the robot

package atlas;

// Ancillary information from the robot

struct status_t
{
  int64_t utime;
  int64_t seq_id;                          
  int64_t processed_to_robot_packet_seq_id;

  int64_t control_data_version_mismatch;

  int32_t run_state;
  int64_t robot_status_flags;

  float pump_inlet_pressure;   
  float pump_supply_pressure;  
  float pump_return_pressure;  
  float air_sump_pressure;     
  float pump_time_meter;     
  float current_pump_rpm;     

  int32_t behavior;

  behavior_feedback_t behavior_feedback;
  stand_feedback_t stand_feedback;
  step_feedback_t step_feedback;
  walk_feedback_t walk_feedback;
  manipulate_feedback_t manipulate_feedback;

  //
  int64_t sensor_head_pps_timestamp;


  //////////////////////////////////////////
  // Run State:
  const int32_t RUN_STATE_IDLE =0;
  const int32_t RUN_STATE_START =1;
  // no state 2
  const int32_t RUN_STATE_CONTROL =3;
  // no state 4
  const int32_t RUN_STATE_STOP =5;

  // Behavior:
  const int32_t BEHAVIOR_NONE=0;        //!< Robot currently not in any behavior; software freeze, controls disabled.
  const int32_t BEHAVIOR_FREEZE=1;      //!< All joints frozen in current position.
  const int32_t BEHAVIOR_STAND_PREP=2;  //!< Robot assumes a balanced posture, preparing for Stand.
  const int32_t BEHAVIOR_STAND=3;       //!< Statically stable stand.
  const int32_t BEHAVIOR_WALK=4;        //!< Dynamically stable walk.
  const int32_t BEHAVIOR_STEP=5;        //!< Statically stable, single-step walk.
  const int32_t BEHAVIOR_MANIPULATE=6;  //!< Statically stable stand, with upper body joints controlled by user.
  const int32_t BEHAVIOR_USER=7;        //!< Setpoints and gains set by user.
  const int32_t BEHAVIOR_CALIBRATE=8;   //!< Robot moves through various calibration motions, returns to Freeze.
  const int32_t BEHAVIOR_SOFT_STOP=9;   //!< "Safe" falling behavior, robot tucks its arms and legs and uses soft position gains
}
